src/rpp/rpp/operators/zip.hpp

```diff
@@ -0,0 +1,129 @@
+//                  ReactivePlusPlus library
+//
+//          Copyright Aleksey Loginov 2023 - present.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          https://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/victimsnino/ReactivePlusPlus
+//
+
+#pragma once
+
+#include <rpp/operators/fwd.hpp>
+
+#include <rpp/defs.hpp>
+#include <rpp/operators/details/combining_strategy.hpp>
+#include <rpp/operators/details/strategy.hpp>
+#include <rpp/schedulers/current_thread.hpp>
+
+#include <list>
+
+namespace rpp::operators::details
+{
+template<rpp::constraint::observer Observer, typename TSelector, rpp::constraint::decayed_type... Args>
+class zip_disposable final : public combining_disposable<Observer, TSelector, Args...>
+{
+public:
+    explicit zip_disposable(Observer&& observer, const TSelector& selector)
+        : combining_disposable<Observer, TSelector, Args...>(std::forward<Observer>(observer), selector)
+    {
+    }
+
+    auto& get_pendings() { return m_pendings; }
+
+private:
+    utils::tuple<std::list<Args>...> m_pendings{};
```

Comment:
could we use dequeue instead? we don't need to insert/remove from middle

Reply:
done